src/main/java/shop/controller/CategoryController.java

```diff
@@ -0,0 +1,81 @@
+package shop.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import shop.dto.request.category.CreateCategoryRequestDto;
+import shop.dto.request.category.UpdateCategoryRequestDto;
+import shop.dto.responce.category.CategoryDto;
+import shop.service.CategoryService;
+
+@RestController
+@RequestMapping(value = "/api/categories", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
You can add `server.servlet.context-path=/api` to application.properties
```suggestion
@RequestMapping(value = "/categories", produces = MediaType.APPLICATION_JSON_VALUE)
```